security/security_tools.md

```diff
@@ -0,0 +1,179 @@
+Report on Dynamic and Static Analysis Security Tools
+====================================================
+
+Recommendations for the CI pipeline
+-----------------------------------
+
+We recommend the following tools for integration in CI for PUL projects:
+
+-   CodeQL
+-   Semgrep
+-   Wapiti
+-   Dastardly
+-   Bearer
+
+Each tool has its own pros and cons, which are detailed in the report below.
+Each project need not implement all of them, most projects would benefit from
+adding one or two to their CI pipelines.
+
+Most of these tools are general: they search for various classes of
+vulnerabilities.  Adding them to your CI pipeline won't provide comprehensive
+protection from any vulnerability, but they will help to:
+* Educate the team about vulnerabilities and mitigation techniques they aren't yet familiar with
+* Enforce secure code patterns
+* Shift security thinking left, from remediation to prevention
+
+The following table gives a brief roundup of the pros and cons to help
+teams decide which tool(s) to start investigating for a particular project:
+
+| Tool | Category | Use case(s) |
+|----|----|----|
+| CodeQL | SAST | A project in Ruby, Java, Javascript, or Python that is looking for a low-barrier setup |
+| Semgrep | SAST | A project in languages that CodeQL doesn't support (e.g. PHP), a project that want to enforce custom rules related to security or code style, or a project that needs CI to run very fast |
+| Wapiti | DAST | A project that wants to use a DAST to check the running application |
+| Dastardly | DAST | A project that wants to use a DAST to check the running application |
```

Comment:
I'm a little unsure why I would want to use Wapitit vs Dastardly or vice versa -- is there anything more we could say here?

Reply:
Below they are listed under different categories. I'm not sure what the difference is between "Dynamic" and "Exploitation Enumeration" -- I guess I would see all of these tools as potentially enumerating exploitation scenarios? Adding a comment to the top recommending a summary of the different categories.

Reply:
I don't have a great sense of when to use Wapiti vs. when to use Dastardly.  @jrgriffiniii and @christinach : any thoughts?

Reply:
I don't have a strong opinion.